lib/util/Util.ts

```diff
@@ -140,6 +140,25 @@ export default class Util {
                         type:        component.type,
                         value:       component.value
                     };
+                } else if (component.type === ComponentTypes.USER_SELECT || component.type === ComponentTypes.ROLE_SELECT || component.type === ComponentTypes.MENTIONABLE_SELECT) {
+                    return {
+                        custom_id:   component.customID,
+                        disabled:    component.disabled,
+                        max_values:  component.maxValues,
+                        min_values:  component.minValues,
+                        placeholder: component.placeholder,
+                        type:        component.type
+                    };
+                } else if (component.type === ComponentTypes.CHANNEL_SELECT) {
+                    return {
+                        channel_types: component.channelTypes,
+                        custom_id:     component.customID,
+                        disabled:      component.disabled,
+                        max_values:    component.maxValues,
+                        min_values:    component.minValues,
+                        placeholder:   component.placeholder,
+                        type:          component.type
+                    };
```

Comment:
Condense these into a singular branch, with an object that can have `channel_types` and `value` added on

(typing the object as Record<string, unknown> or similar will likely be the easiest option, you may need to cast the final result)